src/items.rs

```diff
@@ -2072,20 +2111,40 @@ impl<'a> Rewrite for OpaqueType<'a> {
 
 impl Rewrite for ast::FnRetTy {
     fn rewrite(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<String> {
+        self.rewrite_result(context, shape).ok()
+    }
+    fn rewrite_result(
+        &self,
+        context: &RewriteContext<'_>,
+        shape: Shape,
+    ) -> Result<String, RewriteError> {
         match *self {
-            ast::FnRetTy::Default(_) => Some(String::new()),
+            ast::FnRetTy::Default(_) => Ok(String::new()),
             ast::FnRetTy::Ty(ref ty) => {
                 if context.config.version() == Version::One
                     || context.config.indent_style() == IndentStyle::Visual
                 {
-                    let inner_width = shape.width.checked_sub(3)?;
+                    let inner_width = shape.width.checked_sub(3).ok_or_else(|| {
+                        RewriteError::ExceedsMaxWidth {
+                            configured_width: shape.width,
+                            span: self.span(),
+                        }
+                    })?;
```

Comment:
This isn't something we need to address right now, but I've noticed a lot of `.ok_or_else(|| { RewriteError::ExceedsMaxWidth {..} })` calls throughout the code.

I wonder if we can make this a little more ergonomic. Here are three ideas I have:
1. Define new helper methods on `Shape`
2. Add a macro that will handle some of this boilerplate
3. Define an extension trait so we can more easily convert from `Option<T>` -> `Result<T, RewriteError>`.
   Here's one idea that I had:

   ```rust
   /// Extension trait used to conveniently convert to RewriteError
   pub(crate) trait RewriteErrorExtension<T> {
       fn max_with_error(self, width: usize, span: Span) -> Result<T, RewriteError>;
       fn unknown_error(self) -> Result<T, RewriteError>;
   }

   impl<T> RewriteErrorExtension<T> for Option<T> {
       fn max_with_error(self, width: usize, span: Span) -> Result<T, RewriteError> {
           self.ok_or_else(|| RewriteError::ExceedsMaxWidth {
               configured_width: width,
               span: span,
           })
       }

       fn unknown_error(self) -> Result<T, RewriteError> {
           self.ok_or_else(|| RewriteError::Unknown)
       }
   }
   ```